spring-data-relational/src/main/java/org/springframework/data/relational/core/sql/DefaultSelectBuilder.java

```diff
@@ -320,7 +320,7 @@ public Select build() {
 		@Override
 		public SelectOnConditionComparison on(Expression column) {
 
-			this.from = column;
+			this.from = new Column((Column) column);
```

Comment:
While this solves the problem at hand, I really don't like the looks of it.
1. Creating a `Column` from a `Column` immediately begs the question "Why?".
2. The answer of course is it drops the alias in the process, which is really unexpected when you just look at the constructor.

We therefore should really solve the problem where the SQL is actually generated and make sure that there the unaliased name of the `Column` is used.

Reply:
What would you suggest here? During query generation, I do not have any access to this information, as far that I know.

Reply:
I just had a quick look, so if you think some or all of the following is wrong, please feel free to correct me:

The rendering happens in the {{JoinVisitor}} which in turn delegates it to a {{ConditionVisitor}}. Since the {{ConditionVisitor}} is probably in other places we need a specialised version ({{JoinConditionVisitor}}) which doesn't simply uses {{builder::append}} but checks if it deals with a column and if so uses table + name instead of the alias.

Reply:
I will check that and prepare a new PR. Thank you.

Reply:
I think for this task should be for those that have more experience on this project. 

I still do not see a way to use a Visitor there. At that point, I have only an expression and do not have access to any other value. 

I will close this PR and let someone else work on it.

thank you for your time.